### PR TITLE
[iOS] Implement ItemSizingStrategy hint property for CollectionView

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

Add ItemSizingStrategy to CollectionView in Core; 
Create test harness for ItemSizingStrategy changes; 
Handle ItemSizingStrategy and ItemSizingStrategy changes on iOS;

### Issues Resolved ### 

- partially implements #3172

### API Changes ###

Added:

public enum ItemSizingStrategy;

ItemsView.ItemSizingStrategy // Bindable Property

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Navigate to  CollectionView Gallery -> DataTemplate Galleries -> ItemSizing Strategy, select the different ItemSizingStrategies to see them in action.

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs [hunk 1]
```diff
@@ -220,5 +223,71 @@ public static DataTemplate ScrollToItemTemplate()
...
+
+				image.SetBinding(Image.SourceProperty, new Binding("Image"));
+
+				layout.Content = image;
+
+				return layout;
+			});
+		}
+
+		private class IndexRequestConverter : IValueConverter
```
Comment (samhouts) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]: 👮 
```suggestion
		class IndexRequestConverter : IValueConverter
```

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs [hunk 2]
```diff
@@ -220,5 +223,71 @@ public static DataTemplate ScrollToItemTemplate()
...
+		{
+			private readonly int _cutoff;
```
Comment (samhouts) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]:
```suggestion
			readonly int _cutoff;
```

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs [hunk 3]
```diff
@@ -220,5 +223,71 @@ public static DataTemplate ScrollToItemTemplate()
...
+			private readonly int _lowValue;
```
Comment (samhouts) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]:
```suggestion
			readonly int _lowValue;
```

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs [hunk 4]
```diff
@@ -220,5 +223,71 @@ public static DataTemplate ScrollToItemTemplate()
...
+			private readonly int _highValue;
```
Comment (samhouts) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]:
```suggestion
			readonly int _highValue;
```

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs [hunk 5]
```diff
@@ -220,5 +223,71 @@ public static DataTemplate ScrollToItemTemplate()
...
+			{
+				var index = (int)value;
+
+				return index < _cutoff ? _lowValue : (object)_highValue;
+			}
+
+			public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) => throw new NotImplementedException();
+		}
+
+		private class IndexColorConverter : IValueConverter
```
Comment (samhouts) [inline Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]:
```suggestion
		class IndexColorConverter : IValueConverter
```

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs (hunk 1 above)
Comment (hartez) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]: Fixed.

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs (hunk 2 above)
Comment (hartez) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]: Fixed.

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs (hunk 4 above)
Comment (hartez) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]: Fixed.

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs (hunk 5 above)
Comment (hartez) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]: Fixed.

File: Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs (hunk 3 above)
Comment (hartez) [reply Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/ExampleTemplates.cs]: Fixed.

**Pull request merged**